x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DownsampleStep.java

```diff
@@ -70,16 +67,7 @@ public void performAction(
 
         final String policyName = indexMetadata.getLifecyclePolicyName();
         final String indexName = indexMetadata.getIndex().getName();
-        final String downsampleIndexName = lifecycleState.downsampleIndexName();
-        if (Strings.hasText(downsampleIndexName) == false) {
-            downsampleFailed = true;
-            listener.onFailure(
-                new IllegalStateException(
-                    "downsample index name was not generated for policy [" + policyName + "] and index [" + indexName + "]"
-                )
-            );
-            return;
-        }
+        final String downsampleIndexName = generateDownsampleIndexName(indexName, fixedInterval);
```

Comment:
This just occurred to me - if we have downsampling configured multiple times, say in hot and warm. The hot phase will generate the target index name `downsample-myindex-1h` and then when we get to the warm phase to execute downsampling again we'll end up with the index name `downsample-downsample-myindex-1h-3h` 
(as we don't use the _original_ index name anymore).

This problem exists already in ILM but perhaps becomes a bit more visible with the new naming convention? 
(currently in ILM we can endup with names like this `downsample-8hit-downsample-eruv-.ds-my-data-2023.08.14-000001` )

Perhaps the Downsampling transport action should configure the index provided name setting `index.provided_name` to maintain the original index name? And we can make use of it in ILM and such to generate the new index name based on the originally provided name?

Reply:
Right, this will look strange (and already looks strange). However I don't think the downsample action shouldn't be responsible for this, because the contract of this api is that the target index name is provided. 

I think we should make the `DownsampleStep` smarter and that it generates a better name based on the source index? If the source index doesn't contain a the downsample prefix then it generates the same way it does now in the PR. If the source index contains the downsample index name prefix than it parse out the source index and generates a name from that.

I think we should do this in a follow up PR. This PR is already large and the problem already exists today. I will create a follow up PR and label it as a bug. We can then backport this as well after the FF.

Reply:
Or just use provided index name in DownsampleStep, since current cluster state is available.

Reply:
Great point, thanks for working on this !